cpu/sam0_common/periph/i2c.c

```diff
@@ -92,19 +92,6 @@ void i2c_init(i2c_t dev)
 
     /* I2C using CLK GEN 0 */
     sercom_set_gen(bus(dev),i2c_config[dev].gclk_src);
-#if defined(CPU_FAM_SAML21) || defined(CPU_FAM_SAMR30)
```

Comment:
This seems unrelated. Is there a good reason to remove this part ?

Reply:
What sorcery is this ?

Reply:
Oh right I remember now.
In fact I removed this part on purpose because this is pointless. Sercom slow clock is used only by some SMBus features like timeout. But we don't use them at all, so no need for this clock (and it saves some power).

Reply:
It is unclear to me what this code is about. Are you sure it's strictly not required ? What if someone wants to use the slow clock ?

Reply:
This code setup the SERCOM_GCLK_SLOW clock for the SERCOM your are using. (which is completely unrelated to the SERCOM_GCLK)
According to, SAML21 datasheet, the SERCOM_GLCK_SLOW is used for : 
```
Two generic clocks ared used by SERCOM, GCLK_SERCOMx_CORE and 
GCLK_SERCOM_SLOW.
The core clock (GCLK_SERCOMx_CORE) can clock the I 2 C when working as a master. The slow clock
(GCLK_SERCOM_SLOW) is required only for certain functions, e.g. SMBus timing. These clocks must
be configured and enabled in the Generic Clock Controller (GCLK) before using the I 2 C.

```
Those SMBus features can be active in the CTRLA register for each SERCOM. We're not using them in RIOT codebase.

See chapter 34.5.3 of [SAML21 datasheet](http://ww1.microchip.com/downloads/en/DeviceDoc/60001477A.pdf).

The SERCOM_SLOW clock is NOT used by UART nor SPI.

Reply:
I know it's annoying but I think this change should be in it's own PR because it's not only related to saml1x. In any case, there is no need for this PR to depend on it IIUC.

Reply:
I reapplied this piece of code. (And I will remove it as soon as this PR is merged ;) )